packages/react/src/hooks/use-last-handle.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useMemo, useRef, useCallback } from "react";
+import { Handle } from "../domain/types";
+import { useConfiguration } from "./use-configuration";
+import { useSlashID } from "./use-slash-id";
+
+const STORAGE_LAST_HANDLE_KEY = "@slashid/LAST_HANDLE";
+
+type UseLastHandle = () => {
+  lastHandle: Handle | undefined;
+};
+
+type SuccessEvent = {
+  identifier: Handle | undefined;
+};
+
+export const useLastHandle: UseLastHandle = () => {
+  const { storeLastHandle } = useConfiguration();
+  const { sid, sdkState } = useSlashID();
+  const subscribed = useRef(false);
+
+  const lastHandle = useMemo(() => {
+    const storedHandle = window.localStorage.getItem(STORAGE_LAST_HANDLE_KEY);
+    if (!storeLastHandle || !storedHandle) {
+      return undefined;
+    }
+
+    return JSON.parse(storedHandle);
+  }, [storeLastHandle]);
+
+  const handler = useCallback((e: SuccessEvent) => {
+    window.localStorage.setItem(
+      STORAGE_LAST_HANDLE_KEY,
+      JSON.stringify(e.identifier)
+    );
+  }, []);
+
+  useEffect(() => {
+    if (!storeLastHandle || sdkState !== "loaded" || sid === undefined) {
+      return;
+    }
+
+    if (!subscribed.current) {
+      subscribed.current = true;
+      // @ts-expect-error
```

Comment:
When adding TS annotations like these, it's a good practice to write a comment about the expected error, something like:

```
// @ts-expect-error TODO core SDK does not export the correct event handler type
```

Reply:
This way it's easier for us to come back and fix this later.